### PR TITLE
Fix user registration

### PR DESCRIPTION
**Pull request by savance-ford**
Could you review and test this please? We integrated user login to question posting and filled in some missing user auth stuff. The last files from the views accidentally got included. Don't mind them. Otherwise we can fix it.

### CODE REVIEW

**Comment by jwharrow** [comment]
checking this out now

File: app/helpers/sessions_helpers.rb [hunk 1]
```diff
@@ -6,3 +6,7 @@ def logged_in?
 def current_user
    @_current ||= User.find_by(id: session[:user_id])
```
Comment (jwharrow) [inline app/helpers/sessions_helpers.rb:7]: what's the @_current variable name? that's not something I'm familiar with.

File: app/controllers/users_controllers.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#form for creating new user
+get '/users/new' do
+  erb :_registrationform
```
Comment (jwharrow) [inline app/controllers/users_controllers.rb]: seems strange to make this a partial if it is a stand-alone destination

File: app/models/user.rb [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
   has_many :votes
   has_many :comments
   #add relationship with votes earned.
 
 
-  validates :username, :email, :hashed_password, { presence: true }
+  validates :username, :email, :password, { presence: true }
   validates :username, :email, { uniqueness: true }
   validates :email, format: { with: /\A[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}\z/ }
   validate :check_password
```
Comment (jwharrow) [inline app/models/user.rb]: what happens if raw password is nil? I think line below should probably be `if @raw_password.length == nil || @raw_password.length < 8`

File: app/views/layout.erb [hunk 4]
```diff
@@ -15,7 +15,7 @@
   <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
   <script src="/js/application.js"></script>
 
-  <title></title>
+  <title>DONUT OVERFLOW</title>
```
Comment (jwharrow) [inline app/views/layout.erb:18]: can we add a few of these to the title? `🍩`

**Comment by jwharrow** [comment]
made a few comments to resolve. some just for my own clarity. nothing that should block a merge, so ... 👍

File: app/helpers/sessions_helpers.rb (hunk 1 above)
Comment (savance-ford) [reply app/helpers/sessions_helpers.rb:7]: It's a shortcut to keep us from having to lookup the user every time we call `current_user`. It's saying, if `@_current` doesn't exist, then look them up and set `@_current` to that user.

File: app/views/layout.erb (hunk 4 above)
Comment (savance-ford) [reply app/views/layout.erb:18]: Stretch goal.

**Pull request merged**